omnigan/trainer.py

```diff
@@ -856,6 +879,70 @@ def get_d_loss(self, multi_domain_batch, verbose=0):
         loss = sum(v for d in disc_loss.values() for k, v in d.items())
         return loss
 
+    def update_ADVENT_D(self, multi_domain_batch, verbose=0):
+        # ? split representational as in update_g
+        # ? repr: domain-adaptation traduction
+        self.ADVENT_D_opt.zero_grad()
+        ADVENT_adver_loss = self.get_ADVENT_adver_loss(multi_domain_batch, verbose)
+        ADVENT_adver_loss.backward()
+        ADVENT_D_loss = self.get_ADVENT_D_loss(multi_domain_batch, verbose)
+        ADVENT_D_loss.backward()
+        self.ADVENT_D_opt_step()
+
+        self.logger.losses.ADVENT_discriminator.total_loss = ADVENT_D_loss.item()
+        self.log_losses(model_to_update="ADVENT_D")
+
+    def get_ADVENT_adver_loss(self, multi_domain_batch, verbose=0):
+        """Compute the ADVENT adversarial losses:
+            TODO
+        """
+        for batch_domain, batch in multi_domain_batch.items():
+            x = batch["data"]["x"]
+            z = self.G.encode(x)
+            source_label = 0
+            target_label = 1
+            losses = []
+            lossCal = ADVENTSegLoss(self.opts)
```

Comment:
why `Cal` ? what does it stand for?

Reply:
It means Calculation. Should I use func instead of cal?

Reply:
I think it's a bit unclear to have cal, why not just loss or advent_loss ?

Reply:
Got it. Thanks Meli